<launch>
  <!-- Exclude the .bag from the filename -->
  <arg name="rosbag_filename"/>

  <!-- On computer -->
  <arg name="rosbag_dir" value="/workspace/rosbags/expanded_action_space_study"/>
  <arg name="action_schema_path" value="/workspace/src/feeding_study_cleanup/scripts/detecting_food/data/action_schema_data.csv"/>

  <!-- On harddrive -->
  <!-- <arg name="rosbag_dir" value=""/>
  <arg name="action_schema_path" value="/media/amalnanavati/HCRLAB/2022_11_Bite_Acquisition_Study/extracted_data/action_schema_data_final_2022_05_24.csv"/> -->

  <param name="robot_description" command="cat $(find pr_assets)/data/objects/fork.urdf" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <param name="/use_sim_time" value="true"/>

  <node pkg="tf2_ros" type="static_transform_publisher" name="camera_broadcaster" args="0.00922705  0.00726385  0.01964674 -0.10422086  0.08057444 -0.69519346 0.7061333 CameraBody camera_link" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="fork_broadcaster" args="0.13347163 -0.02011249  0.1598728 -0.13918934  0.2998962  -0.50758908 0.79561947 ForqueBody fork_tip" />

  <node pkg="feeding_study_cleanup" type="publish_transforms.py" name="publish_transforms"/>

  <node pkg="rosbag" type="play" name="play" args="--clock $(arg rosbag_dir)/$(arg rosbag_filename).bag --delay=1 --rate=0.2"/> <!-- output="screen"  -->

  <node pkg="feeding_study_cleanup" type="visualize_action_schema.py" name="visualize_action_schema" output="screen">
    <param name="rosbag_filename" type="string" value="$(arg rosbag_filename)"/>
    <param name="action_schema_filepath" type="string" value="$(arg action_schema_path)"/>
  </node>

  <!-- Detected food poses for 2-sandwich-5_2021-11-23-15-08-52.bag -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="detected_food_0_broadcaster" args="-0.401092952273 0.0173056617831 0.208050119336 -0.566155015612 -0.442743463958 -0.413554149171 0.558945157526 TableBody detected_food_0" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="detected_food_1_broadcaster" args="-0.460070724696 0.0118304243272 0.0903235603244 -0.643304338113 -0.320488385352 -0.293332367581 0.630398957446 TableBody detected_food_1" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="detected_food_2_broadcaster" args="-0.473250644296 0.0130185303596 0.136794797049 0.479088726847 0.535749258879 0.505745535077 -0.477145865715 TableBody detected_food_2" />
  <node pkg="tf2_ros" type="static_transform_publisher" name="detected_food_3_broadcaster" args="-0.425705604135 0.0152065080822 0.164526785744 -0.717982679594 0.0324676513567 0.04993176159 0.6935081417 TableBody detected_food_3" /> -->

</launch>
